### PR TITLE
remove deprecated mod_http_notifications

### PR DESCRIPTION
**Pull request by janciesla8818**
This PR removes deprecated `mod_http_notifications`. The functionality is operated by `mod_event_pusher` with `http` backend. The changes include:
* remove of the `mod_http_notifications`
* rename `mod_http_notifications_SUITE` to `mod_event_pusher_http_SUITE`
* update tests in `mod_event_pusher_http_SUITE` 
* update metrics names from `mod_http_notification` to `mod_event_pusher_http`

### CODE REVIEW

**Comment by mongoose-im** [comment]
[8645.1](https://travis-ci.org/esl/MongooseIM/jobs/736086042) / Erlang 23.0.3 / small_tests / 44b2f2a31ce7658487efce10a662f665321eca44
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2912/8645/small_tests.23.0.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2912/8645/small_tests.23.0.3/small/ct_run.mongooseim@localhost.2020-10-15_15.56.08/lib.mongooseim.logs/run.2020-10-15_15.56.09/suite.log.html)

**Comment by NelsonVides** [comment]
This should probably need some notice in the migration notes, you could create a new 4.0.0-4.x.x.md file and mention that this module has been removed as it was deprecated a long time ago in version x.y.z, and that the metric name has changed accordingly.

**Comment by mongoose-im** [comment]
[8650.1](https://travis-ci.org/esl/MongooseIM/jobs/736411042) / Erlang 23.0.3 / small_tests / 750fe540da3c171952632fedeb3d933620016b68
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2912/8650/small_tests.23.0.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2912/8650/small_tests.23.0.3/small/ct_run.mongooseim@localhost.2020-10-16_15.52.05/lib.mongooseim.logs/run.2020-10-16_15.52.06/suite.log.html)

File: doc/migrations/4.0.0_4.X.X.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+### `mod_http_notification` module is no longer available
+
+`mod_http_notification` has been deprecated since MongooseIM 2.1.1 and it is no longer available in this release.
+Please use the `http` backend for `mod_event_pusher`, which is the direct equivalent of `mod_http_notification`.
+
+For more details on how to configure `mod_event_pusher` with `http` backend, please see [this section](..//modules/mod_event_pusher_http/).
```
Comment (NelsonVides) [inline doc/migrations/4.0.0_4.X.X.md:6]: Isn't it correct like:
```suggestion
For more details on how to configure `mod_event_pusher` with `http` backend, please see [this section](../modules/mod_event_pusher_http/).
```

**Review by NelsonVides — COMMENTED** [review]
Seems fine except for perhaps an unnecessary character, and maybe mentioning that the metric name also changed. Most likely unimportant, but somebody could be possibly annoyed to rely on a metric name that just disappeared (the old metric name disappeared, right?)

File: doc/migrations/4.0.0_4.X.X.md (hunk 1 above)
Comment (janciesla8818) [reply doc/migrations/4.0.0_4.X.X.md:6]: Good catch! 💪

**Review by NelsonVides — APPROVED** [review]
Excellent, thanks for the work! 😄
Now just waiting for CI and we can merge 🙂

**Comment by mongoose-im** [comment]
[8653.1](https://travis-ci.org/esl/MongooseIM/jobs/737028378) / Erlang 23.0.3 / small_tests / 8f6bc7266e70add45a0d462e53d5acdd42b1d03e
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2912/8653/small_tests.23.0.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2912/8653/small_tests.23.0.3/small/ct_run.mongooseim@localhost.2020-10-19_11.26.35/lib.mongooseim.logs/run.2020-10-19_11.26.35/suite.log.html)

**Pull request merged**